pkg/deps/jar_test.go

```diff
@@ -0,0 +1,148 @@
+//
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package deps_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/apache/skywalking-eyes/license-eye/pkg/deps"
+)
+
+func TestCanResolveJarFile(t *testing.T) {
+	resolver := new(deps.JarResolver)
+	for _, test := range []struct {
+		fileName string
+		exp      bool
+	}{
+		{"1.jar", true},
+		{"/tmp/1.jar", true},
+		{"1.jar2", false},
+		{"protobuf-java-3.13.0.jar", true},
+		{"slf4j-api-1.7.25.jar", true},
+	} {
+		b := resolver.CanResolve(test.fileName)
+		if b != test.exp {
+			t.Errorf("JarResolver.CanResolve(\"%v\") = %v, want %v", test.fileName, b, test.exp)
+		}
+	}
+}
+
+func copyJars(t *testing.T, pomFile, content string) ([]string, error) {
+	dir := filepath.Dir(pomFile)
+
+	if err := os.Chdir(dir); err != nil {
+		return nil, err
+	}
+
+	if err := dumpPomFile(pomFile, content); err != nil {
+		return nil, err
+	}
+
+	if _, err := exec.Command("mvn", "dependency:copy-dependencies", "-DoutputDirectory=./lib", "-DincludeScope=runtime").Output(); err != nil {
```

Comment:
This is not portable, users / developers don't have `mvn` installed won't be able to run this test

Reply:
The purpose of this test is to check the usability of the JarResolver. Can I put the jar files directly in the test/testdata?

Reply:
> The purpose of this test is to check the usability of the JarResolver. Can I put the jar files directly in the test/testdata?

you can put binary files (`.jar`) files in the codebase but we can't release them, this is unusual case, maybe we can only run this test case when `mvn` is available, otherwise we skip this test?

Reply:
> 
> you can put binary files (`.jar`) files in the codebase but we can't release them, this is unusual case, maybe we can only run this test case when `mvn` is available, otherwise we skip this test?

ok, i will check the availability of mvn first